pillarbox-player/src/main/java/ch/srgssr/pillarbox/player/session/MediaSessionCallbackImpl.kt

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (c) SRG SSR. All rights reserved.
+ * License information is available from the LICENSE file.
+ */
+package ch.srgssr.pillarbox.player.session
+
+import android.os.Bundle
+import android.util.Log
+import androidx.media3.common.MediaItem
+import androidx.media3.exoplayer.SeekParameters
+import androidx.media3.session.MediaLibraryService
+import androidx.media3.session.MediaSession
+import androidx.media3.session.MediaSession.MediaItemsWithStartPosition
+import androidx.media3.session.SessionCommand
+import androidx.media3.session.SessionCommands
+import androidx.media3.session.SessionError
+import androidx.media3.session.SessionResult
+import ch.srgssr.pillarbox.player.PillarboxExoPlayer
+import ch.srgssr.pillarbox.player.PillarboxPlayer
+import ch.srgssr.pillarbox.player.session.PillarboxMediaSession.Callback
+import ch.srgssr.pillarbox.player.utils.DebugLogger
+import com.google.common.util.concurrent.Futures
+import com.google.common.util.concurrent.ListenableFuture
+
+internal open class MediaSessionCallbackImpl(
+    val callback: Callback,
+    val mediaSession: PillarboxMediaSession
+) : MediaSession.Callback {
+
+    override fun onConnect(session: MediaSession, controller: MediaSession.ControllerInfo): MediaSession.ConnectionResult {
+        val availableSessionCommands = SessionCommands.Builder().apply {
+            if (session is MediaLibraryService.MediaLibrarySession) {
+                addSessionCommands(MediaSession.ConnectionResult.DEFAULT_SESSION_AND_LIBRARY_COMMANDS.commands)
+            } else {
+                addSessionCommands(MediaSession.ConnectionResult.DEFAULT_SESSION_COMMANDS.commands)
+            }
+            // TODO maybe add a way integrators can add custom commands
+            val player = session.player
+            if (player is PillarboxPlayer) {
+                addSessionCommands(PillarboxSessionCommands.AVAILABLE_COMMANDS)
+                if (player.isSeekParametersSupported) {
+                    addSessionCommands(listOf(PillarboxSessionCommands.COMMAND_GET_SEEK_PARAMETERS))
+                }
+                if (player is PillarboxExoPlayer) {
```

Comment:
Do we need this check, since `PillarboxPlayer` provides the `setImageOutput()` method?